packages/calcite-components/src/components/tile/tile.tsx

```diff
@@ -89,6 +172,52 @@ export class Tile implements InteractiveComponent {
 
   @Element() el: HTMLCalciteTileElement;
 
+  private clickHandler = (): void => {
+    this.setFocus();
+    this.handleSelectEvent();
+  };
+
+  private containerEl: HTMLDivElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * @internal
+   */
+  @Event({ cancelable: false }) calciteInternalTileKeyEvent: EventEmitter<KeyboardEvent>;
+
+  /**
+   * Fires when the selected state of the component changes.
+   */
+  @Event() calciteTileSelect: EventEmitter<void>;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private handleSelectEvent = (): void => {
+    if (this.disabled) {
+      return;
+    }
+    if (!this.interactive) {
+      return;
+    }
+    if (this.selectionMode === "single-persist" && this.selected === true) {
```

Comment:
This and the above conditions all return early. Please merge them into the same condition.